fdbclient/NativeAPI.actor.cpp

```diff
@@ -815,24 +832,69 @@ Future<Void> DatabaseContext::onConnected() {
 	return cluster->onConnected();
 }
 
+ACTOR static Future<Void> switchConnectionFileImpl(Reference<ClusterConnectionFile> connFile, DatabaseContext* self) {
+	TEST(true); // Switch connection file
+	TraceEvent("SwitchConnectionFile")
+	    .detail("ConnectionFile", connFile->canGetFilename() ? connFile->getFilename() : "")
+	    .detail("ConnectionString", connFile->getConnectionString().toString());
+
+	// Reset state from former cluster.
+	self->masterProxies.clear();
```

Comment:
I believe it is possible to get a new clientInfo from the current proxies, which will then let the self-masterProxies get re-populated with proxies from the old cluster file. I am considering reverting the ability to forward from one proxy generation to the next, but as of the current state of master I believe would be a problem.

Reply:
By triggering `masterProxiesChangeTrigger` after clearing the proxies, this situation is avoided now. Every time `clientInfo` would be changed in response to a reply from a proxy, `onMasterProxiesChanged()` causes the reply to be ignored (e.g. in the case of `tryCommit`) or the corresponding request to be retried (with the updated set of proxies) (e.g. in the case of `getConsistentReadVersion`).

Reply:
We've now run another 1000 test runs (with the `DifferentClustersSameRV` workload) after this last change, and all have passed.